arbitrator/prover/src/kzgbn254.rs

```diff
@@ -45,21 +46,23 @@ pub fn prove_kzg_preimage_bn254(
 
     let mut kzg = KZG.clone();
 
-    // expand the roots of unity, should work as long as it's longer than chunk length and chunks
-    // from my understanding the data_setup_mins pads both min_chunk_len and min_num_chunks to 
-    // the next power of 2 so we can load a max of 2048 from the test values here
-    // then we can take the roots of unity we actually need (len polynomial) and pass them in
-    // @anup, this is a really gross way to do this, pls tell better way
-    kzg.data_setup_mins(1, 2048)?;
+    // expand roots of unity
+    let roots_of_unity = kzg.calculate_roots_of_unity(preimage.len() as u64);
+    match roots_of_unity {
+        Ok(_) => (),
+        Err(err) => return Err(err.into()),
+    };
```

Comment:
```suggestion
    let roots_of_unity = kzg.calculate_roots_of_unity(preimage.len() as u64)?;
```

Or if you need to wrap the error:
```suggestion
    let roots_of_unity = kzg.calculate_roots_of_unity(preimage.len() as u64).map_err(|e| Err(e))?;
```